### PR TITLE
apm-1131: use service id instead of service_name

### PR DESCRIPTION
**Pull request by vurt007**

### CODE REVIEW

File: ansible/roles/create-api-deployment-pre-reqs/templates/terraform/locals.tf [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 locals {
   account_id     = data.aws_caller_identity.current.account_id
   region         = data.aws_region.current.name
   aws_ecr_bucket = "prod-eu-west-2-starport-layer-bucket"
 
-  env_service_name       = "${var.apigee_environment}-${var.service_name}"
-  service_namespaces     = var.apigee_environment == "internal-dev" ? [var.service_name, "${var.service_name}-pr-*"] : [var.service_name]
+  env_service_id       = "${var.apigee_environment}-${var.service_id}"
+  service_namespaces     = var.apigee_environment == "internal-dev" ? [var.service_id, "${var.service_id}-*"] : [var.service_id]
+  short_service_namespaces     = var.apigee_environment == "internal-dev" ? [var.service_id, "${var.service_id}-*"] : [var.service_id]
```
[the remark this answers is not in the document]
Comment (vurt007) [reply ansible/roles/create-api-deployment-pre-reqs/templates/terraform/locals.tf:11]: ah yeah .. not required anymore .. everything is a 'short_service_namespace'

File: ansible/roles/destroy-api-deployment-dns-entry/tasks/main.yml [hunk 2]
```diff
@@ -34,9 +34,14 @@
       with_filetree: "{{ '../templates' }}"
       when: item.state == 'file'
 
+    - name: terraform init
+      shell: "make -C {{ out_dir }}/terraform clean init" # noqa 305
```
[the remark this answers is not in the document]
Comment (vurt007) [reply ansible/roles/destroy-api-deployment-dns-entry/tasks/main.yml:38]: no .. for some reason these Make commands aren't working as 'command:'  not been able to figure out why yet

File: ansible/roles/create-api-deployment-pre-reqs/vars/main.yml [hunk 3]
```diff
@@ -1,5 +1,3 @@
 ---
-workspace: "api-deployment:{{ account }}:{{ apigee_environment }}:{{service_name}}"
+workspace: "api-deployment:{{ account }}:{{ apigee_environment }}:{{ service_id }}"
```
[the remark this answers is not in the document]
Comment (vurt007) [reply ansible/roles/create-api-deployment-pre-reqs/vars/main.yml:2]: yes .. in this the 'pre-reqs' are not pr namespaced ...  ( so we don't have to create new users / pull request )

File: ansible/roles/deploy-ecs-proxies/vars/main.yml [hunk 4]
```diff
@@ -1,9 +1,11 @@
 ---
+
 build_label: "{{ lookup('env', 'build_label') }}"
-images: "{{ docker_service | json_query('[].name') | unique | sort }}"
 service_base_path: "{{ SERVICE_BASE_PATH | regex_replace('^\\/|\\/$', '') }}"
-sub_workspace: "{{ ':pr-' + pr_number if pr_number else '' }}"
-workspace: "api-deployment:{{ account }}:{{ apigee_environment }}:{{service_name}}{{ sub_workspace }}"
+sub_workspace: "{{ ':' + pr_number if pr_number else '' }}"
```
[the remark this answers is not in the document]
Comment (vurt007) [reply ansible/roles/deploy-ecs-proxies/vars/main.yml:5]: yes, we do similar in 'setup-facts' role ... think there's a bunch of streamlining on this to be done on this, but will pick that up in a later ticket

File: ansible/roles/destroy-ecs-proxies/tasks/main.yml [hunk 5]
```diff
@@ -17,19 +17,23 @@
     {{ aws_cmd }} ecs list-tasks --cluster apis-{{ apigee_environment }}
     --service {{ namespaced_name }} --query taskArns
   register: list_tasks
+  failed_when: list_tasks.rc != 0 and 'ServiceNotActiveException' not in list_tasks.stderr and 'ServiceNotFoundException' not in list_tasks.stderr
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply ansible/roles/destroy-ecs-proxies/tasks/main.yml:20]: Ignore me, we've restricted to a particular service above. If we get a ServiceNotFoundException for one task, we'll get it for all

**Pull request merged**